shared/js/background/dnr-gpc.js

```diff
@@ -0,0 +1,54 @@
+import settings from './settings'
+import tdsStorage from './storage/tds'
+import { GPC_HEADER_RULE_ID } from './dnr-utils'
+import { GPC_HEADER_PRIORITY } from '@duckduckgo/ddg2dnr/lib/rulePriorities'
+const {
+    resourceTypes,
+    generateDNRRule
+} = require('@duckduckgo/ddg2dnr/lib/utils')
+
+/**
+ * Ensure that the rule to add Global Privacy Control (GPC) request headers is
+ * added when the feature is enabled, and removed when the feature is disabled.
+ * @param {Object} config
+ *   The privacy configuration.
+ * @return {Promise}
+ */
+export async function ensureGPCHeaderRule (config = null) {
+    const removeRuleIds = [GPC_HEADER_RULE_ID]
+    const addRules = []
+
+    if (!config) {
+        await tdsStorage.ready('config')
+        config = tdsStorage.config
+    }
+
+    const gpcEnabled = settings.getSetting('GPC') &&
+          config?.features?.gpc?.state === 'enabled'
+
+    if (gpcEnabled) {
+        const allowedDomains = config.features.gpc.exceptions?.map(e => e.domain)
+
+        addRules.push(
+            generateDNRRule({
+                id: GPC_HEADER_RULE_ID,
+                priority: GPC_HEADER_PRIORITY,
+                actionType: 'modifyHeaders',
+                requestHeaders: [
+                    { header: 'Sec-GPC', operation: 'set', value: '1' }
+                ],
+                resourceTypes: [...resourceTypes],
+                excludedInitiatorDomains: allowedDomains,
+                excludedRequestDomains: allowedDomains
+            })
+        )
+    }
+
+    await chrome.declarativeNetRequest.updateSessionRules({
+        removeRuleIds, addRules
+    })
+}
+
+settings.onSettingUpdate.addEventListener(
+    'GPC', () => { ensureGPCHeaderRule() }
+)
```

Comment:
Let's wrap this in a component - this will allow us to avoid using globals (`settings` and `tdsStorage` can be passed to the component constructor), and avoid calling `settings.onSettingsUpdate` on module import. This will make it easy for us to write a simple unittest for this component.